cmd/kubelet/app/server.go

```diff
@@ -803,6 +804,14 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 			return fmt.Errorf("topology manager policy options %v require feature gates %q enabled",
 				s.TopologyManagerPolicyOptions, features.TopologyManagerPolicyOptions)
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.NodeSwap) {
+			if !isCgroup2UnifiedMode() && s.MemorySwap.SwapBehavior == kubelettypes.LimitedSwap {
```

Comment:
Shouldn't we return an error since this is not supported? What do we want to let kubelet continue running for an unsupported combination?

Reply:
Thats a good point. We never failed in the cases around this swap feature so it has become difficult to enforce that from this point.

We had an issue to fail cgroupv1 and this feature but then we discovered that people were using `--fail-swap-on=false` with cgroupv1. So I may have been more pragmatic here than I needed.

Reply:
Users will need to explicitly specify `LimitedSwap` in this case, so failing cleanly may be a better experience

Reply:
I think this PR is minutes away from merging so can i do a minor follow up?

Reply:
follow up to fail fast at startup when explicitly configured to use limited swap on a machine where it cannot sgtm

Reply:
follow up sounds good :+1:

Reply:
This PR is taking its sweet time to merge..  

Its like watching water boil.

I have https://github.com/kubernetes/kubernetes/pull/123738 as a quick follow up once this merges. I'll check and rebase accordingly when this merges.

Reply:
SG. Thanks

Reply:
https://github.com/kubernetes/kubernetes/pull/123738 is up now and rebased.